### PR TITLE
SC2: Added Difficulty Override to Client

### PR DESCRIPTION
**Pull request by TheCondor07**
Simple command to the client override the difficulty of a missions in a certain seed.  Is not persistent and is local only, will only have the override until the client is closed

### CODE REVIEW

**Comment by Alchav** [comment]
Yay, I won't have to do this by hacking the source anymore 😅

File: Starcraft2Client.py [hunk 1]
```diff
@@ -38,6 +38,30 @@
...
     ctx: SC2Context
 
+    def _cmd_difficulty(self, difficulty: str = "") -> bool:
+        """Overrides the current difficulty set for the seed.  Takes the argument casual, normal, hard, or brutal"""
+        options = difficulty.split()
+        num_options = len(options)
+        set_difficulty = True
+
+        if num_options > 0:
+            if options[0] == "Casual" or "casual":
```
Comment (Berserker66) [inline Starcraft2Client.py:48]: you can do new_text = text.lower() to get all lowercase, to ignore case.

Comment (TheCondor07) [reply Starcraft2Client.py:48]: Thanks, will implement.

File: Starcraft2Client.py [hunk 2]
```diff
@@ -38,6 +38,31 @@
...
+            if difficulty_choice == "casual":
+                self.ctx.difficulty_override = 0
+            elif difficulty_choice == "normal":
+                self.ctx.difficulty_override = 1
+            elif difficulty_choice == "hard":
+                self.ctx.difficulty_override = 2
+            elif difficulty_choice == "brutal":
+                self.ctx.difficulty_override = 3
+            else:
+                sc2_logger.info("Unable to parse difficulty '" + options[0] + "'")
```
Comment (Berserker66) [inline Starcraft2Client.py:58]: please use self.output instead of sc2_logger

File: Starcraft2Client.py [hunk 3]
```diff
@@ -38,6 +38,31 @@
...
+                set_difficulty = False
+
+            if set_difficulty:
+                sc2_logger.info("Difficulty set to " + options[0])
+        else:
+            sc2_logger.info("Difficulty needs to be specified in the command.")
```
Comment (Berserker66) [inline Starcraft2Client.py:64]: function should return True if succesful, False if failed.

**Review by Berserker66 — APPROVED** [review]

**Comment by TheCondor07** [comment]
Alright, this should address both your comments, @Berserker66.  Also went through the play and disable mission check commands to add returns that were missing.

**Pull request merged**